app/main.go

```diff
@@ -236,6 +236,15 @@ func Main(config Config) {
 	configureLogging(cli, ctx.Command(), p)
 
 	config.State.LockTimeout = cli.getLockTimeout()
+	// Note. Set default sources before applying additional sources so that the
+	// defaulting behavior doesn't change depending on if additional sources are applied.
+	// The default sources shouldn't disappear if additional sources are supplied.
+	if config.State.Sources == nil {
```

Comment:
Instead of pulling this out of the State constructor, I'd prefer to push the additional logic _into_ the constructor. Perhaps via an additional field on the State config.

Reply:
I made a few changes. The scope expanded a little beyond what you were originally getting at.

There was some confusion in my mind about what `state.Config` was supposed to represent. Is it an input to `Open(...)` that is immutable? Does it contain the state after it's stored in the object?

I "solved" my confusion, by making a choice. It's now an immutable input to `Open(...)` and is not retained in the `State` object.

WDYT?